exercises/stack_04.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+
+var log = console.log;
+var assert = console.assert;
+
+window.Collections = Object.create(null);
+
+(function(root) {
+    function Stack(arr=[]) {
+        this.arr = arr;
+    };
+
+    //stack.forEach
+    Stack.prototype.forEach = function(callback) {
+        for (var i = 0; i < this.arr.length; i++) {
+            callback(this.arr[i], i, this.arr);
+        }
+    };
+
+    // stack.push
+    Stack.prototype.push = function(addition) {
+        this.arr[this.arr.length] = addition;
+    };
+    
+    //stack.peek
+    Stack.prototype.peek = function() {
+        return this.arr[this.arr.length - 1];
+    };
+
+    //stack.pop
+    Stack.prototype.pop = function() {
+        return this.arr.splice(-1, 1);
+    };
+
+    //stack.clear
+     Stack.prototype.clear = function() { 
+        while (this.arr.length > 0) {
+            this.pop();
+        }
+    };
+
+   Stack.prototype.extend = function(obj) {
+       for (var i = 0; i < obj.arr.length; i++) {
+            this.push(obj.arr[i]);
+       }
+    };
+
+    //stack.copy
+    Stack.prototype.copy = function() {
+        var newStack = new Collections.Stack();
```

Comment:
Don't need to use `Collections.` here, since `Stack` is available in scope.

Reply:
👍